### PR TITLE
Update iproj.json and update/create .ibmi.json

### PR DESCRIPTION
**Pull request by irfanshar**

### CODE REVIEW

File: src/makei/init_project.py [hunk 1]
```diff
@@ -168,40 +177,95 @@ def create_file(file_path: Path, content: Optional[str], force: bool = False) ->
...
+        else:
+            version = retrieve_json_val(iproj_json_path, "version")
+
+            if objlib is not None:
+                # Update iproj.json
+                update_json_field(iproj_json_path, "objlib", objlib)
+
+                # Update .ibmi.json
+                if os.path.exists(ibmi_json_path):
+                    update_json_field(ibmi_json_path, "build", objlib, "objlib")
```
[the remark this answers is not in the document]
Comment (edmundreinhardt) [reply src/makei/init_project.py:227]: In fact I just tested and the .ibmi.json is ignored if both iproj.json and .ibmi.json's objlib are specified

[the remark this answers is not in the document]
Comment (edmundreinhardt) [reply src/makei/init_project.py:227]: Please remove .ibmi.json/objlib

File: src/makei/init_project.py [hunk 2]
```diff
@@ -168,40 +177,95 @@ def create_file(file_path: Path, content: Optional[str], force: bool = False) ->
...
+    except KeyError:
+        returnVal = None
 
-    proj_spec = ProjSpec()
+    return returnVal
+
+
+def init_project(force: bool = False, objlib: str = "", tgtCcsid: str = "") -> None:
+    """ A CLI utility to create a project"""
+    signal.signal(signal.SIGINT, _signal_handler)
```
[the remark this answers is not in the document]
Comment (irfanshar) [reply src/makei/init_project.py:208]: To handle when the user enters Ctrl+C, and to inform them that the makei init was cancelled

File: src/makei/ibmi_json.py [hunk 3]
```diff
@@ -49,7 +49,7 @@ def from_file(cls, file_path: Path, parent_ibm_i_json: "IBMiJson") -> "IBMiJson"
     def __dict__(self):
         build = {}
 
-        if self.build["tgt_ccsid"] != DEFAULT_TGT_CCSID and self.build["tgt_ccsid"] != "":
+        if self.build["tgt_ccsid"] != DEFAULT_TGT_CCSID and self.build["tgt_ccsid"] != None:
```
Comment (edmundreinhardt) [inline src/makei/ibmi_json.py:52]: probably should guard against either, if it is possible to store "" as a value in Json then we should not consider the value specified

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]
see my many comments on the objlib commit

File: src/makei/init_project.py [hunk 4]
```diff
@@ -168,40 +175,88 @@ def create_file(file_path: Path, content: Optional[str], force: bool = False) ->
         file.write(content)
 
 
-def init_project(force: bool = False) -> None:
-    """ A CLI utility to create a project"""
-    signal.signal(signal.SIGINT, _signal_handler)
+def update_json_field(pathToJsonFile: str, updateKey1: str, updateVal: str, updateKey2: str = "") -> None:
```
Comment (edmundreinhardt) [inline src/makei/init_project.py:178]: So what are the semantices.  IF the value is "" do we not add the key-value pair?  What about if it is undefined?
Is there a way to delete the field if the `undefined` or empty string is specified
We should really standardize on whether "" or None means unspecified.

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]

File: src/makei/build.py [hunk 5]
```diff
@@ -94,7 +94,7 @@ def _create_build_vars(self):
         subdirs = list(map(lambda x: x.parents[0], rules_mk_paths))
 
         subdirs.sort(key=lambda x: len(x.parts))
-        dir_var_map = {Path('.'): IBMiJson.from_values(self.iproj_json.tgt_ccsid, self.iproj_json.objlib)}
+        dir_var_map = {Path('.'): IBMiJson.from_values(self.iproj_json.tgt_ccsid)}
```
[the remark this answers is not in the document]
Comment (edmundreinhardt) [reply src/makei/build.py]: @irfanshar did you respond to this comment

[the remark this answers is not in the document]
Comment (irfanshar) [reply src/makei/build.py]: @edmundreinhardt Yes, my mistake, I restored objlib being passed in

**Review by edmundreinhardt — APPROVED** [review]
Looks good

**Pull request merged**